src/window.py

```diff
@@ -772,6 +789,13 @@ def _on_delete_failed(self, worker, path):
     def _on_popup_closed(self, button, popup, data):
         self._clean_popups()
 
+    def _on_button_closed_go_home(self, button):
+        self._reset_to_path(PortfolioPlaces.PORTFOLIO_HOME_DIR)
+        self._on_button_closed(button)
+        # Revert the change of the callback
+        self.close_button.handler_block(self.handler_close_go_home)
```

Comment:
Instead of doing this, you can just return  GLib.SOURCE_REMOVE.

So it would look like

```
self.close_button.handler_unblock(self.default_close_handler)
return GLib.SOURCE_REMOVE
```